### PR TITLE
Issue Command

### PR DESCRIPTION
**Pull request by sourTaste000**
Todos:
- [x] Fetching an issue from github's api with token
- [x] Sends message to chat if the token is misisng
- [x] Parses Json
- [x] Creates an embed with all information and sending it to the chat
- [x] Add labels
- [x] Support pulls
- [x] Add assignees
- [ ] Search issues by words
- [ ] Make a tree that defaults to kami-blue for user

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -25,6 +26,7 @@ dependencies {
     compile "com.mojang:brigadier:$bot_brigadier_version"
     compile "org.slf4j:slf4j-simple:$bot_slf4j_version"
     compile "com.github.ronmamo:reflections:$bot_reflections_version"
+    implementation "com.beust:klaxon:$bot_claxon_version"
```
Comment (dominikaaaa) [inline build.gradle:29]: why is this needed?

File: src/main/kotlin/classes/IssueClass.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+data class IssueClass(
```
Comment (dominikaaaa) [inline src/main/kotlin/classes/IssueClass.kt:1]: use more consistent naming for the rest, they should all either have the Class suffix or none at all

File: src/main/kotlin/commands/IssueCommand.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import doesLater
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import string
+
+/**
+ * @author sourTaste000
+ * @since 2020/9/8
+ */
+object IssueCommand : Command("issue") {
```
Comment (dominikaaaa) [inline src/main/kotlin/commands/IssueCommand.kt:19]: add help usage

**Review by dominikaaaa — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 4]
```diff
@@ -11,6 +11,7 @@ version "$bot_kt_version"
 
 repositories {
     mavenCentral()
+    jcenter()
```
Comment (dominikaaaa) [inline build.gradle:14]: you can remove this as well

File: src/main/kotlin/commands/IssueCommand.kt [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    init {
+        string("user") {
```
Comment (dominikaaaa) [inline src/main/kotlin/commands/IssueCommand.kt:21]: Make a tree that defaults to kami-blue for user

File: src/main/kotlin/commands/IssueCommand.kt [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            string("repoName") {
+                string("issueNum") {
+                    doesLater { context ->
+                        val user: String = context arg "user"
+                        val repoName: String = context arg "repoName"
+                        val issueNum: String = context arg "issueNum"
+                        val githubToken = FileManager.readConfigSafe<AuthConfig>(ConfigType.AUTH, true)?.githubToken
```
Comment (dominikaaaa) [inline src/main/kotlin/commands/IssueCommand.kt:28]: this should return and send an error in the channel if the token is null (see rules command for a concise oneliner example)

**Review by dominikaaaa — CHANGES_REQUESTED** [review]

File: gradle/wrapper/gradle-wrapper.properties [hunk 7]
```diff
@@ -1,6 +1,5 @@
-#Sun Aug 16 16:43:20 EDT 2020
```
Comment (dominikaaaa) [inline gradle/wrapper/gradle-wrapper.properties:1]: why are we touching these?

File: gradlew [hunk 8]
```diff
@@ -82,6 +82,7 @@ esac
```
Comment (dominikaaaa) [inline gradlew:82]: see above

File: gradlew.bat [hunk 9]
```diff
@@ -29,6 +29,9 @@ if "%DIRNAME%" == "" set DIRNAME=.
 set APP_BASE_NAME=%~n0
```
Comment (dominikaaaa) [inline gradlew.bat:29]: see above

File: src/main/kotlin/commands/IssueCommand.kt [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+                                        "https://kamiblue.org",
+                                        "https://cdn.discordapp.com/avatars/743237292294013013/591c1daf9efcfdd7ea2db1592d818fa6.png"
+                                    )
+                                    url = result.html_url
+                                }
+                            }
+                        }catch(e: Exception){
+                            message.channel.send{
+                                embed{
+                                    title = "Error!"
```
Comment (dominikaaaa) [inline src/main/kotlin/commands/IssueCommand.kt:84]: do just "Error", to be more consistent with other commands

File: src/main/kotlin/commands/IssueCommand.kt [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+                                    description = "Something went wrong when trying to execute this command! Does the user/repo/issue exist?"
```
Comment (dominikaaaa) [inline src/main/kotlin/commands/IssueCommand.kt:85]: Maybe print the actual stacktrace?

**Review by dominikaaaa — CHANGES_REQUESTED** [review]

**Comment by sourTaste000** [comment]
very very basic issue command that only shows the title, the description and the state. pulls, assignees and labels are planned for later.

**Comment by sourTaste000** [comment]
now supports labels

**Comment by sourTaste000** [comment]
when merge uwu

**Comment by humboldt123** [comment]
when merge my [\#17](https://github.com/kami-blue/assets/pull/17)

**Comment by sourTaste000** [comment]
mfw

**Comment by sourTaste000** [comment]
> when merge my [\#17](https://github.com/kami-blue/assets/pull/17)

Pixel art cringe

**Review by dominikaaaa — CHANGES_REQUESTED** [review]

**Comment by sourTaste000** [comment]
fuck forgot to push

**Comment by sourTaste000** [comment]
Done. Ready for merge uwu

File: src/main/kotlin/commands/IssueCommand.kt [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+ * @since 2020/9/8
+ */
+@Suppress("BlockingMethodInNonBlockingContext")
+object IssueCommand : Command("issue") {
+    init {
+        /*
+        TODO:
+             Make a tree that defaults to kami-blue for user
+             Support Multiple Assignees
+         */
```
Comment (dominikaaaa) [inline src/main/kotlin/commands/IssueCommand.kt:28]: Is this resolved?

File: src/main/kotlin/commands/IssueCommand.kt [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+                                message.channel.send {
+                                    embed {
+                                        title = "[$name]"
+                                        description = "Github Token not found!"
+                                        color = Main.Colors.ERROR.color
+                                    }
+                                }
+                                return@doesLater
+                            }
+                        val request = Request.Builder().addHeader("Authorization", "token $githubToken").url("https://api.github.com/repos/$user/$repoName/issues/$issueNum").get().build()
```
Comment (dominikaaaa) [inline src/main/kotlin/commands/IssueCommand.kt:47]: image
This format doesn't look right

File: src/main/kotlin/commands/IssueCommand.kt [hunk 14]
```diff
@@ -0,0 +1,128 @@
...
+                                    e.printStackTrace()
+                                    color = Main.Colors.ERROR.color
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
```
Comment (dominikaaaa) [inline src/main/kotlin/commands/IssueCommand.kt:119]: I'll have to format this myself

**Review by dominikaaaa — CHANGES_REQUESTED** [review]

File: src/main/kotlin/commands/IssueCommand.kt (hunk 12 above)
Comment (sourTaste000) [reply src/main/kotlin/commands/IssueCommand.kt:28]: whops

File: src/main/kotlin/commands/IssueCommand.kt (hunk 13 above)
Comment (sourTaste000) [reply src/main/kotlin/commands/IssueCommand.kt:47]: i tried to fetch /rate_limit with that, works correctly

File: src/main/kotlin/commands/IssueCommand.kt (hunk 14 above)
Comment (sourTaste000) [reply src/main/kotlin/commands/IssueCommand.kt:119]: ok

**Comment by sourTaste000** [comment]
The only thing that is bad is sending two requests if it is a pull

**Comment by sourTaste000** [comment]
is this the format u wanted lol

**Comment by dominikaaaa** [comment]
No, I'll format it myself later. I'm reviewing it today

**Comment by sourTaste000** [comment]
k

**Comment by sourTaste000** [comment]
when merge :weary:

**Comment by dominikaaaa** [comment]
Sorry, god damnit I thought I was force pushing our branch. Intellij git sucks sometimes. I'll make a new pull
